Fase-1/API/self-order-management/src/main/kotlin/com/fiap/selfordermanagement/application/services/OrderService.kt

```diff
@@ -1,39 +1,248 @@
+import com.fiap.selfordermanagement.application.ports.incoming.PlaceOrderUseCase
+import com.fiap.selfordermanagement.application.ports.incoming.PrepareOrderUseCase
+import com.fiap.selfordermanagement.application.ports.incoming.ProvidePaymentRequestUseCase
+import com.fiap.selfordermanagement.application.ports.incoming.SyncPaymentStatusUseCase
 import com.fiap.selfordermanagement.application.ports.outgoing.OrderRepository
+import org.springframework.transaction.annotation.Transactional
+import java.time.LocalDate
 
-class OrderService(
-    private val repository: OrderRepository,
-) : CancelOrderUseCase, CompleteOrderUseCase, GetOrdersUseCase, DeleteItemsUseCase, UpsertItemsUseCase,
-    UpsertOrderUseCase {
-    override fun cancelOrder(orderId: Long): Order? {
-        return repository.findById(orderId)?.let {
-            cancelOrder(order = it)
+open class OrderService(
+    private val orderRepository: OrderRepository,
+    private val getCustomersUseCase: LoadCustomerUseCase,
+    private val getProductUseCase: LoadProductUseCase,
+    private val getStockUseCase: LoadStockUseCase,
+    private val adjustInventoryUseCase: AdjustInventoryUseCase,
+    private val loadPaymentUseCase: LoadPaymentUseCase,
+    private val providePaymentRequestUseCase: ProvidePaymentRequestUseCase,
+    private val syncPaymentStatusUseCase: SyncPaymentStatusUseCase,
+) : LoadOrderUseCase,
+    PlaceOrderUseCase,
+    IntentOrderPaymentUseCase,
+    ConfirmOrderUseCase,
+    PrepareOrderUseCase,
+    CompleteOrderUseCase,
+    CancelOrderStatusUseCase {
+    override fun getByOrderNumber(orderNumber: Long): Order {
+        return orderRepository.findByOrderNumber(orderNumber)
+            ?: throw SelfOrderManagementException(
+                errorType = ErrorType.ORDER_NOT_FOUND,
+                message = "Order $orderNumber not found",
+            )
+    }
+
+    override fun findAll(): List<Order> {
+        return orderRepository.findAll()
+    }
+
+    override fun findByStatus(status: OrderStatus): List<Order> {
+        return orderRepository.findByStatus(status)
+    }
+
+    // TODO: expose in endpoint
+    override fun findByCustomerNickname(nickname: String): List<Order> {
+        return orderRepository.findByCustomerNickname(nickname)
+    }
+
+    // TODO: expose in endpoint
+    override fun findByCustomerNicknameAndStatus(
+        nickname: String,
+        status: OrderStatus,
+    ): List<Order> {
+        return orderRepository.findByCustomerNicknameAndStatus(nickname, status)
+    }
+
+    // TODO: expose in endpoint
+    override fun findByCustomerDocument(document: String): List<Order> {
+        return orderRepository.findByCustomerDocument(document)
+    }
+
+    // TODO: expose in endpoint
+    override fun findByCustomerDocumentAndStatus(
+        document: String,
+        status: OrderStatus,
+    ): List<Order> {
+        return orderRepository.findByCustomerDocumentAndStatus(document, status)
+    }
+
+    @Transactional
```

Comment:
Por favor, verifique quando usar transação ou não, prestando atenção quando exceptions são lançadas, e se algum efeito colateral é intencionalmente persistido, como marcar um pedido como `REJECTED` se pagamento não for encontrado (not found exception).